var/spack/repos/builtin/packages/py-traitlets/package.py

```diff
@@ -25,7 +26,8 @@ class PyTraitlets(PythonPackage):
 
     depends_on('python@3.7:', when='@5:', type=('build', 'run'))
     depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
-    depends_on('py-setuptools@40.8:', type='build')
+    depends_on('py-hatchling@0.25:', when='@5.2.1.post0:', type='build')
+    depends_on('py-setuptools@40.8:', when='@:5.2.1.a', type='build')
```

Comment:
Is this a typo on the version in the `when` clause (i.e., is `.a` deliberate)?

Reply:
The `.a` is deliberate. No 5.2.1.a release exists, but `@:5.2.1` includes `5.2.1.post0` so we need to append a `.a` to cap the range before then.